### DISCUSSION TITLE
Gotek SFRKC30.AT2

### DISCUSSION THREAD

**Discussion by Jmabp** [keirf/flashfloppy#775] [General]
Hi,

I got a Gotek SFRKC30.AT2 ( this model on picture ) that I try to make work with an Akai CD3000 but the problem I got is there's no screen / buttons or rotary.

![DSC01651](https://user-images.githubusercontent.com/129078108/227961817-cbfb5d10-e1c6-4981-8853-f5f83762b741.JPG)
![DSC01652](https://user-images.githubusercontent.com/129078108/227961832-0812d672-6e26-4d6d-8cff-89eb731505cb.JPG)

Is there a way to know if it was flashed or to flash it myself (knowing that there is no button controller, screen or rotary fader as mentioned above)?

I tinkered with a USB A to USB A cable but when I plug it into my pc it is recognized as an unsigned USB device, and when I use the Arteryprogrammer.exe software to try to recognize it (after installing the drivers as the chip is an Artery it seems to me cf: see photo), it does not recognize it in USB DFU,

So I'm looking for a lead if anyone has an idea.

Thanks

**Comment by tenuke** [comment]
I think that you should know based on sellers description.
Where did you buy from it?
If there was zero indication of it being anything but what it is straight
from factory, no reasons to assume it perhaps having any another than
standard firmware

ma 27. maalisk. 2023 klo 17.05 Jmabp ***@***.***> kirjoitti:

**Reply by Jmabp** [reply]
Thank you for your help, I got it from a friend who bought an EMU sampler on which it was installed (but not well connected elsewhere), so it's second hand and I have no idea if it was flashed (FF or hxc) and since I don't have an Oled screen.

Do you think it is possible under these conditions (imagining that it is in standard firmware) to flash it anyway without screen and without buttons?

**Reply by keirf** [reply]
Save yourself the pain and get a Gotek with a display. Preferably OLED. And while you're at it get a "AT32F435" Gotek which is the latest sort with more RAM. Is arsing about with a display-less Gotek really worth your time?

**Comment by Jmabp** [comment]
Thank you keirf for your answer, I actually intend to either take an Oled in the long term with buttons (since the PCB allows me to add them, it seems to me) and I try because I have no other means of using my sampler for the moment (Akai CD3000 model) no "rec" input and if there is no screen and I can "emulate" just a floppy disk for example I would be happy because I could start having fun with making the link with my PC (waiting to possibly have a more complete Gotek), but is it possible? It does look complicated...

**Reply by keirf** [reply]
The problem is that when it doesn't work, you have no feedback about why at all.

**Reply by Jmabp** [reply]
Yes indeed, but I managed to get the led on by plugging it into my Akai and it makes a USB connection sound on windows so I think it's possible to make it work, I don't have a lot of money right now to get another one so i'm trying to see if i can make it work but if you have good experience in gotek emulators and flashing and installing firmwares and tell me: forget it, it might be much too hard indeed, I'll take your opinion into consideration :d

**Reply by keirf** [reply]
If you can get the device recognised by Windows in programming mode then just flash FlashFloppy using the straightforward USB-DFU instructions. https://github.com/keirf/FlashFloppy/wiki/Firmware-Programming

Follow instructions for "Artery Microcontrollers" and make sure to use the at415 firmware, not at435!

**Comment by Jmabp** [comment]
Thanks for your support keirf, but I encounter difficulties that make me believe that I cannot manage to make this gotek work

By following the procedure explained on the HxC site here: https://hxc2001.com/docs/gotek-floppy-emulator-hxc-firmware/pages/flash-a-gotek-with-a-usb-cable.html# artery-usb-fw-installation

After having deduced that the chip was an Artery (it is small maybe I was wrong about it?), I do the USB A procedure:

Connections of pins 1 & 2 (seamless ensuring that there is no bad contact).

![DSC01653](https://user-images.githubusercontent.com/129078108/228022715-ded5d441-5be9-465d-9a2c-aec7d5cecccc.JPG)

I download the DfuSe software from ARTERY.

I install the drivers.

I connect with USB A the pcb (the windows sound of a USB connection works).

And I see that ( The Gotek not appear ) :

![USB DFU](https://user-images.githubusercontent.com/129078108/228022753-27ec2ff4-2a4f-4058-9f9c-810c240f1447.jpg)

The device has no signature (despite the drivers installed).

![gotek gestionnaire de périphériques](https://user-images.githubusercontent.com/129078108/228022751-7bb7455b-cbb0-4bb1-affb-bf169b185a81.jpg)

And it is not recognized in the Artery ISP programmer software (provided on the HxC site), in the USB DFUs it does not appear.

So I tried with the more recent version of Artery same problem, I looked at the English manual which gives a track at the level of the drivers by recommending to install via the device manager manually the drivers what I tried to do :

![driver 1](https://user-images.githubusercontent.com/129078108/228022757-de71e29c-25a9-4d82-b1df-9077cd4c704b.jpg)

![driver 2](https://user-images.githubusercontent.com/129078108/228022759-72778a9f-27c4-4237-96fc-49d3aa6067a8.jpg)

(excerpt above from the Artery English manual).

The procedure I tried to follow:

![driver 3](https://user-images.githubusercontent.com/129078108/228022763-27e7a7df-f00a-4281-9e44-72ec6159e145.jpg)

I don't have quite the same url when I put a link to the drivers in the manual to the Artery folder as in the manual, the mention is missing: "usb_driver" and I can't find it in the folder provided on HxC nor on the latest official Artery update so I have this message at the end:

![driver 4](https://user-images.githubusercontent.com/129078108/228022766-2dad1e3e-f937-4b8a-bf79-bcfe0f1a9734.jpg)

And therefore impossible to flash and continue the procedure, knowing that as I have no screen or button or rotary fader but possibilities are also reduced, I still hope to do something with this little Gotek box (if it's possible without an oled screen), here you go while waiting to find other leads, thank you for reading.

Axel

**Comment by keirf** [comment]
What version of Windows are you running? Before looking for the folder 'usb_driver', did you 'Extract driver' in the Artery_DFU_DriverInstall program?

**Comment by Jmabp** [comment]
Win 10 version - 21H2

And actually no I had not extracted the drivers and in addition to that by manipulating the wire that I had put to make the connection to launch the Gotek in "boot" mode I realized that now it does not appear as a device without signature but as it should in "AT32 Bootloader DFU"

![driver 5](https://user-images.githubusercontent.com/129078108/228027158-b4e78ce3-1f5f-45a3-af2c-8c13b3c8c1ed.jpg)

The drivers are now correctly extracted

![driver 6](https://user-images.githubusercontent.com/129078108/228027150-be4bd42f-1222-43a4-a48d-a1166d51b8de.jpg)

I will try to do the manipulation indicated in the manual to install them manually and then return to the USB DFU detection step with the Artery program ISP (I'm using the latest version)

Thanks a lot for theses advices keirf,

Axel
